cardano-cli/test/cardano-cli-test/Test/Cli/Parser.hs

```diff
@@ -0,0 +1,99 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cli.Parser
+  ( hprop_integral_reader
+  , hprop_integral_pair_reader_positive
+  , hprop_integral_pair_reader_negative
+  )
+where
+
+import           Cardano.CLI.EraBased.Options.Common (integralParsecParser,
+                   pairIntegralParsecParser)
+
+import           Data.Bits (Bits)
+import           Data.Data (Proxy (..), Typeable)
+import           Data.Either (isLeft, isRight)
+import           Data.Word (Word16)
+import qualified Text.Parsec as Parsec
+
+import           Hedgehog (Gen, Property, assert, property)
+import           Hedgehog.Extras (propertyOnce)
+import qualified Hedgehog.Gen as Gen
+import           Hedgehog.Internal.Property (forAll)
+import qualified Hedgehog.Range as Gen
+import qualified Hedgehog.Range as Range
+
+-- | Execute me with:
+-- @cabal test cardano-cli-test --test-options '-p "/integral reader/"'@
+hprop_integral_reader :: Property
+hprop_integral_reader = property $ do
+  assert $ isRight $ parse @Word "0"
+  assert $ isRight $ parse @Word "42"
+  assert $ isLeft $ parse @Word "-1"
+  assert $ isLeft $ parse @Word "18446744073709551616"
```

Comment:
It would be better to write this as
```suggestion
  assertWith isLeft $ parse @Word "18446744073709551616"
```
`assert` does not report the value on assertion failure

Reply:
Changed to (for example) `assertWith (parse @Word "18446744073709551616") isLeft`